tgui/packages/tgui/interfaces/Wizard_Spellbook.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author IPingu (https://github.com/IPling)
+ * @license ISC
+ */
+
+import { useBackend, useLocalState, useSharedState } from '../backend';
+import { Box, Button, Collapsible, Dimmer, Flex, Input, LabeledList, Section, Stack } from '../components';
+import { Window } from '../layouts';
+
+// This will do in-place of there being no wizard theme
+const category_coloring = (spell_category) => {
+  switch (spell_category) {
+    case "Enchantment":
+      return "purple";
+    case "Equipment":
+      return "yellow";
+    case "Offensive":
+      return "red";
+    case "Defensive":
+      return "blue";
+    case "Utility":
+      return "green";
+    case "Miscellaneous":
+      return "grey";
+  }
+};
+
+export const Wizard_Spellbook = (props, context) => {
+  const { data } = useBackend(context);
+  const [searchQuery, setSearchQuery] = useLocalState(context, 'searchQuery', '');
+  const {
+    spellbook_contents,
+    spell_slots,
+    owner_name,
+  } = data;
+
+  let spell_categories = [];
+  for (let spell_category in spellbook_contents) {
+    spell_categories.push(spell_category);
+  }
+
+  return (
+    <Window
+      title={"Wizard Spellbook"}
+      theme={"ntos"}
+      fontSize={2}
+      height={600}
+      width={500}
+    >
+      <Window.Content>
+        <Section title={owner_name+"'s Spellbook "}>
+          <Flex justify={"space-between"}>
+            <Flex.Item>
+              {"Spell slots remaining: "+spell_slots}
+            </Flex.Item>
+            <Flex.Item>
+              <Input
+                value={searchQuery}
+                placeholder={"search by spell name"}
```

Comment:
```suggestion
                placeholder={"Search by spell name"}
```
capitalization might be nice imo